qiskit_experiments/analysis/data_processing.py

```diff
@@ -47,9 +47,27 @@ def filter_data(data: List[Dict[str, any]], **filters) -> List[Dict[str, any]]:
 
 
 def mean_xy_data(
-    xdata: np.ndarray, ydata: np.ndarray, sigma: Optional[np.ndarray] = None, method: str = "sample"
+    xdata: np.ndarray,
+    ydata: np.ndarray,
+    sigma: Optional[np.ndarray] = None,
+    method: str = "sample",
 ) -> Tuple[np.ndarray]:
-    r"""Return (x, y_mean, sigma) data.
+    r"""Wrapper function for multi_mean_xy_data
+    in the case of one data series
+    """
+    series = np.zeros(xdata.size)
+    x_means, y_means, y_sigmas, _ = multi_mean_xy_data(xdata, ydata, sigma, series, method)
```

Comment:
I think the code would be clearer if this function was left unchanged, and the multi function called this function on each series and concatenated the results. This would make it easier to add additional methods in the future without having to also deal with indexing be series.

eg something like:

```python
def multi_mean_xy_data(
    series: np.ndarray, xdata: np.ndarray, ydata: np.ndarray,
    sigma: Optional[np.ndarray] = None, method: str = "sample"):
    
    series_vals = np.unique(series)
    
    series_means = []
    xdata_means = []
    ydata_means = []
    sigma_means = []

    # Get x, y, sigma data for series and process mean data
    for i in series_vals:
        idxs = series == series_vals[i]
        sigma_i = sigma[idxs] if sigma is not None else None
        x_mean, y_mean, sigma_mean = mean_xy_data(
            xdata[idxs], ydata[idxs], sigma=sigma_i, method=method)
        series_means.append(i * np.ones(x_mean.size, dtype=int))
        xdata_means.append(x_mean)
        ydata_means.append(y_mean)
        sigma_means.append(sigma_mean)
    
    # Concatenate lists
    return (
        np.concatenate(series_means),
        np.concatenate(xdata_means),
        np.concatenate(ydata_means),
        np.concatenate(sigma_means))
```